packages/bot-skeleton/src/scratch/blocks/Binary/Trade Definition/trade_definition_tradeoptions.js

```diff
@@ -522,7 +522,7 @@ Blockly.Blocks.trade_definition_tradeoptions = {
     setCurrency() {
         const currency_field = this.getField('CURRENCY_LIST');
         const { currency } = DBotStore.instance.client;
-        currency_field?.setText(getCurrencyDisplayCode(currency));
+        currency_field.setValue(getCurrencyDisplayCode(currency));
```

Comment:
@rupato-deriv let's keep the optional chaining just in case if `currency_field` is undefined.

```suggestion
        currency_field?.setValue(getCurrencyDisplayCode(currency));
```